core/src/main/scala/akka/persistence/r2dbc/internal/PubSub.scala

```diff
@@ -55,11 +56,27 @@ import akka.persistence.typed.PersistenceId
     val slice = persistenceExt.sliceForPersistenceId(pid)
 
     val offset = TimestampOffset(timestamp, timestamp, Map(pid -> pr.sequenceNr))
+    val notTagged =
+      pr.payload match {
+        case Tagged(payload, _) =>
+          // eventsByTag not implemented (see issue #82), but events can still be tagged, so we unwrap this tagged event.
+          PersistentRepr(
```

Comment:
use this instead: `pr.withPayload(payload)`

Reply:
but this copy of the `PersistentRepr` isn't needed at all. This match can extract the `Tagged(payload, _)` or `pr.payload` and then when creating the EventEnvelope use that extracted payload instead of `Option(pr.payload)`